.github/workflows/builder.yml

```diff
@@ -54,18 +51,37 @@ jobs:
   #                       Build the builder                         #
   #                                                                 #
   ###################################################################
+  detect-env:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # Needed to detect the current reusable repository and ref.
+      contents: read
+    outputs:
+      builder_repo: ${{ steps.detect.outputs.builder_repo }}
+      builder_ref: ${{ steps.detect.outputs.builder_ref }}
+    steps:
+      - name: Detect the builder ref
+        id: detect
+        shell: bash
+        run: | 
+          export OIDC_TOKEN=$(curl -sS -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value')
```

Comment:
although it's unlikely to fail in practice, can we add some error handling, something like https://github.com/ossf/scorecard-action/blob/36a22b695be1a46ca0e2bc47549cd4d311f6c0ee/entrypoint.sh#L50-L55?

Reply:
done!